src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -153,6 +153,9 @@ namespace QgsWms
     renderer.drawLegend( context );
     painter.end();
 
+    // restore layers
+    restorer.reset( new QgsWmsRestorer( mContext ) );
+    // return
```

Comment:
I'm not sure to understand what you're trying to achieve. The `restorer` is a unique pointer so the destructor is called once out of scope. And it's actually in the destructor that layers are restored to their original state.

So unless I'm missing something, I'd said that it's not useful.

Reply:
@pblottiere If you take alook at the test I have added, without an explicite reset:
1. GetStyles contains only QGIS styles
2. GetMap with SLD_BODY
3. GetStyles contains QGIS styles and the UserStyle of the GetMap with SLD_BODY

Reply:
@pblottiere I have found the issue and fix it.